### ISSUE TITLE
Training time a bit slow

### ISSUE THREAD

**Issue by DuckSQ-BIT** [2U1/Qwen2-VL-Finetune#172]
Thanks for open-sourcing this great work! However, I’ve encountered an issue with training speed. I’m using Qwen2 as the backbone for my project, but I found that running finetune.sh took 40 minutes for just 313 steps. This means that fine-tuning for 20,000 steps would take around 42.6 hours, which feels quite slow. I’d like to know if there are any ways to accelerate the process.

<img width="1757" height="97" alt="Image" src="https://github.com/user-attachments/assets/bbfdec25-9669-4b5b-bcae-4e8c65550e07" />

**Comment by 2U1** [comment]
This could be the issue from 
1. Using offloading
2. `use_reentrant=True`

For the first one you could avoid use the script witout offloading.
For the second one, I'm not quite sure for the reason but, it has a but that when using `use_reentrant=False` with gradient checkpointing.

**Comment by 2U1** [comment]
@DuckSQ-BIT When you are training with deepspeed zeor3, you could change some settings in the deepspeed config.
```
    "reduce_scatter": true,
    "reduce_bucket_size": 5e7,
    "stage3_prefetch_bucket_size": 5e7,
    "stage3_param_persistence_threshold": 1e5,
    "stage3_max_live_parameters": 1e9,
    "stage3_max_reuse_distance": 1e9,
```
Something like this would be a bit faster but it will take more vram.
